File: Python Labs/com/pattern/decorator/ClothingItemDecorator.py
from com.pattern.decorator.ClothingItem import ClothingItem


class ClothingItemDecorator(ClothingItem):
    def __init__(self, decorated_item):
        self.decorated_item = decorated_item

    def get_description(self):
        return self.decorated_item.get_description()

    def get_price(self):
        return self.decorated_item.get_price()

File: Java Labs/Patterns On Java/src/com/pattern/decorator/ClothingItemDecorator.java
package com.pattern.decorator;

public abstract class ClothingItemDecorator implements ClothingItem {
    protected ClothingItem decoratedItem;

    public ClothingItemDecorator(ClothingItem decoratedItem) {
        this.decoratedItem = decoratedItem;
    }

    @Override
    public String getDescription() {
        return decoratedItem.getDescription();
    }

    @Override
    public double getPrice() {
        return decoratedItem.getPrice();
    }
}
